src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -1414,6 +1419,31 @@ static MixMonoToStereoF GetMonoToStereoMixer(const WAVEFORMATEXTENSIBLE *fmtext,
     return NULL;
 }
 
+// ------------------------------------------------------------------------------------------
+static HRESULT TryAllocateMonoMixerBuffer(PaWasapiSubStream *subStream, UINT32 frames)
+{
+    assert(subStream->monoMixer != NULL);
+
+    // If not provided then use max buffer size of WASAPI device
+    if (frames == 0)
+        frames = subStream->bufferSize;
+
+    assert(frames != 0);
+
+    // Expand buffer if necessary
+    UINT32 monoBufferSize = frames * subStream->wavexu.ext.Format.nBlockAlign;
+    if (monoBufferSize > subStream->monoBufferSize)
+    {
+        subStream->monoBuffer = PaWasapi_ReallocateMemory(subStream->monoBuffer, (subStream->monoBufferSize = monoBufferSize));
```

Comment:
assignment inside function call parameter expression seems unnecessarily hard to follow.

Reply:
Yes, but this prevent accidental misuse of unassigned `subStream->monoBufferSize` in `PaWasapi_ReallocateMemory`. I do not mind splitting it though as per your proposal.